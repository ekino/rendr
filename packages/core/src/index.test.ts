import { createPage } from "./index";

describe("test create page", () => {
  it("empty page", () => {
    const page = createPage({});

    expect(page).toMatchSnapshot();
  });

  it("with head", async () => {
    const page = createPage({
      head: {
        title: "My Custom Title",
        meta: []
      }
    });

    expect(page).toMatchSnapshot();
  });

  it("with cache", () => {
    const page = createPage({
      cache: {
        ttl: 999
      }
    });

    expect(page).toMatchSnapshot();
  });

  it("with block definitions", () => {
    const page = createPage({
      blocks: [
        {
          container: "main",
          settings: {
            foo: "bar"
          },
          type: "my.container"
        },
        {
          settings: {
            non: "valid block as missing the type attribute"
          }
        },
        1
      ]
    });

    expect(page).toMatchSnapshot();
  });
});
