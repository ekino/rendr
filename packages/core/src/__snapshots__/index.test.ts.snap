// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test create page empty page 1`] = `
Page {
  "blocks": Array [],
  "cache": Object {
    "ttl": 0,
  },
  "head": Object {
    "defaultTitle": "-",
    "htmlAttributes": Object {},
    "link": "",
    "meta": Array [],
    "title": "-",
    "titleTemplate": "%s",
  },
  "id": "id",
  "path": "",
  "settings": Object {},
  "statusCode": 200,
  "template": "default",
  "type": "document",
}
`;

exports[`test create page with block definitions 1`] = `
Page {
  "blocks": Array [
    Object {
      "container": "main",
      "order": 0,
      "settings": Object {
        "foo": "bar",
      },
      "type": "my.container",
    },
  ],
  "cache": Object {
    "ttl": 0,
  },
  "head": Object {
    "defaultTitle": "-",
    "htmlAttributes": Object {},
    "link": "",
    "meta": Array [],
    "title": "-",
    "titleTemplate": "%s",
  },
  "id": "id",
  "path": "",
  "settings": Object {},
  "statusCode": 200,
  "template": "default",
  "type": "document",
}
`;

exports[`test create page with cache 1`] = `
Page {
  "blocks": Array [],
  "cache": Object {
    "ttl": 999,
  },
  "head": Object {
    "defaultTitle": "-",
    "htmlAttributes": Object {},
    "link": "",
    "meta": Array [],
    "title": "-",
    "titleTemplate": "%s",
  },
  "id": "id",
  "path": "",
  "settings": Object {},
  "statusCode": 200,
  "template": "default",
  "type": "document",
}
`;

exports[`test create page with head 1`] = `
Page {
  "blocks": Array [],
  "cache": Object {
    "ttl": 0,
  },
  "head": Object {
    "defaultTitle": "-",
    "htmlAttributes": Object {},
    "link": "",
    "meta": Array [],
    "title": "My Custom Title",
    "titleTemplate": "%s",
  },
  "id": "id",
  "path": "",
  "settings": Object {},
  "statusCode": 200,
  "template": "default",
  "type": "document",
}
`;

exports[`test createContext test clientSide context 1`] = `
Object {
  "asPath": "https://ekino.com/foobar?foo=bar",
  "hostname": "ekino.com",
  "isClientSide": true,
  "isServerSide": false,
  "params": Object {},
  "pathname": "/foobar",
  "query": Object {
    "foo": "bar",
  },
  "req": null,
  "res": null,
  "settings": Object {},
}
`;

exports[`test createContext test clientSide context 2`] = `
Object {
  "asPath": "https://ekino.com/foobar?foo=bar",
  "hostname": "ekino.com",
  "isClientSide": true,
  "isServerSide": false,
  "params": Object {},
  "pathname": "/foobar",
  "query": Object {
    "foo": "bar",
  },
  "req": null,
  "res": null,
  "settings": Object {
    "foo": "bar",
  },
}
`;

exports[`test mergePages should merge properly 1`] = `
Page {
  "blocks": Array [
    Object {
      "container": "header",
      "order": 1,
      "settings": Object {},
      "type": "type-parent-1",
    },
    Object {
      "container": "header",
      "order": 1,
      "settings": Object {},
      "type": "type-parent-2",
    },
    Object {
      "container": "header",
      "order": 1,
      "settings": Object {},
      "type": "type-child-1",
    },
    Object {
      "container": "header",
      "order": 1,
      "settings": Object {},
      "type": "type-child-2",
    },
  ],
  "cache": Object {
    "ttl": 100,
  },
  "head": Object {
    "defaultTitle": "The default title",
    "htmlAttributes": Object {
      "foo": "bar",
    },
    "link": "",
    "meta": Array [
      Object {
        "name": "foo",
      },
      Object {
        "property": "value",
      },
    ],
    "title": "-",
    "titleTemplate": "%s",
  },
  "id": "the child page",
  "path": "/the-child",
  "settings": Object {},
  "statusCode": 403,
  "template": "2-columns",
  "type": "document",
}
`;
